### PR TITLE
Makes locking movement a hotkey

### LINKED ISSUE #50216: control+WASD still roots you in place, even if you've changed your bindings
Reproduction steps:
1. Bind both WASD and control+WASD to moving.
2. Bind both your arrow keys and control+arrow keys to rotating in place.
3. Press control while moving.
4. Notice that you're rooted in place (you won't rotate because you unbound control+WASD from rotating in place, so you'll instead just remain facing the direction you were moving in until you let go of control), so your cyborg-optimized keybind setup doesn't actually work.
5. Cry as you realize that you'll forever more have to stop in place to attempt to grab people or bolt doors behind you as a cyborg.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
code: Moves locking movement from being a hardcoded thing to being a customizable hotkey. The default for this is now Alt, moving movement toggle to C
/:cl:

still need to look into a bug that existed before this PR that causes the key to be let go even though you were holding it.

this pr was made in response to some of the comments in #53755

Fixes #50216

### CODE REVIEW

**Comment by Rohesie** [comment]
Fixes #50216

**Comment by Qustinnus** [comment]
it does fix that, unless I misunderstood the issues.

**Comment by Rohesie** [comment]
If you hold `ctrl` while running, to click something, you'll stop where you are.
This does not seem to fix that.

**Review by Rohesie — DISMISSED** [review]

File: code/datums/keybinding/mob.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 
 /datum/keybinding/mob/face_north
-	hotkey_keys = list("CtrlW", "CtrlNorth")
+	hotkey_keys = list("W", "North")
```
Comment (Rohesie) [inline code/datums/keybinding/mob.dm:7]:
```suggestion
	hotkey_keys = list("AltW", "AltNorth")
```

File: code/datums/keybinding/mob.dm [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 
 /datum/keybinding/mob/face_east
-	hotkey_keys = list("CtrlD", "CtrlEast")
+	hotkey_keys = list("D", "East")
```
Comment (Rohesie) [inline code/datums/keybinding/mob.dm:23]:
```suggestion
	hotkey_keys = list("AltD", "AltEast")
```

File: code/datums/keybinding/mob.dm [hunk 3]
```diff
@@ -36,7 +36,7 @@
 
 
 /datum/keybinding/mob/face_south
-	hotkey_keys = list("CtrlS", "CtrlSouth")
+	hotkey_keys = list("S", "South")
```
Comment (Rohesie) [inline code/datums/keybinding/mob.dm:39]:
```suggestion
	hotkey_keys = list("AltS", "AltSouth")
```

File: code/datums/keybinding/mob.dm [hunk 4]
```diff
@@ -51,7 +51,7 @@
 	return TRUE
 
 /datum/keybinding/mob/face_west
-	hotkey_keys = list("CtrlA", "CtrlWest")
+	hotkey_keys = list("A", "West")
```
Comment (Rohesie) [inline code/datums/keybinding/mob.dm:54]:
```suggestion
	hotkey_keys = list("AltA", "AltWest")
```

File: code/datums/keybinding/mob.dm (hunk 1 above)
Comment (Qustinnus) [reply code/datums/keybinding/mob.dm:7]: yeah this was there as i was tryna fix the issue we discussed

**Comment by Tharcoonvagh** [comment]
Dont make it alt, its used to see the contents of a tile, I suggest changing it to something like pause, since locking movement is a niche thing that not many people use or know about imo

**Comment by Rohesie** [comment]
> 
> 
> Dont make it alt, its used to see the contents of a tile, I suggest changing it to something like pause, since locking movement is a niche thing that not many people use or know about imo

You can rebind it to something else if you want.
Alt makes sense, if you want to see the contents of a tile in general you'll stop moving as well.

**Review by Rohesie — APPROVED** [review]

**Comment by Qustinnus** [comment]
what roh said was my thought process, noone runs and checks a tile

**Comment by Tharcoonvagh** [comment]
Okay then, when I tried to bind stuff to alt and ctrl it refused to work, guess ill try again

**Comment by ATH1909** [comment]
> what roh said was my thought process, noone runs and checks a tile

No, but they do run and alt+click an extinguisher to extinguisher slip people, or run and alt+click a door to shock it as a silicon, or hold down alt to temporarily walk around on walk intent.

Still, THANK YOU for this PR, it'll be a GODSEND for diehard borg mains. I'm just worried that, due to the key that you chose to move it to, it's gonna be quite annoying for anyone who HASN'T reconfigured their hotkeys yet.

**Comment by ATH1909** [comment]
Wait, why have you kept the FaceNorth, FaceSouth, etc. hotkeys around? Aren't they now entirely redundant with this new movement locking hotkey?

**Comment by ATH1909** [comment]
Also, isn't there some weird stuff you have to do whenever you update the preferences menus? Like, you have to update a version or something? I ran into that with my PR that added extra quickdraw hotkeys.

**Comment by Rohesie** [comment]
> 
> 
> Also, isn't there some weird stuff you have to do whenever you update the preferences menus? Like, you have to update a version or something? I ran into that with my PR that added extra quickdraw hotkeys.

No, they are just not updated automatically. Which means the users will have to either assign them manually or reset to hotkey mode if they don't have custom changes.

**Comment by Angustmeta** [comment]
Tested this now. 

The 'turn north' etc bindings causes problems. 

To bind 'block movemen't to other keys than ctrl/alt/shift you must be bind the "turn" keys to WASD, but that makes you turn oddly while walking diagonally.

**Comment by Rohesie** [comment]
> The 'turn north' etc bindings cause problems.
> 
> To bind 'block movement' to other keys than ctrl/alt/shift you must bind the "turn" keys to WASD, but that makes you turn oddly while walking diagonally.

Could you expand on this / clarify?
What happens if you bind blocking movement to something like space? Do you still move?
Why do you need to bind facing to WASD to bind blocking movement to keys other than ctrl/alt/shift?

**Comment by ATH1909** [comment]
> Tested this now.
> 
> The 'turn north' etc bindings cause problems.
> 
> To bind 'block movement' to other keys than ctrl/alt/shift you must bind the "turn" keys to WASD, but that makes you turn oddly while walking diagonally.

I've binded "block movement" to u (and then c) with no issues. Could you show us a video or something?

**Pull request merged**